dashboard/app/controllers/maker_controller.rb

```diff
@@ -121,18 +121,26 @@ def login_code
   # GET /maker/display_code
   # renders a page for users to copy and paste a login key
   def display_code
-    # Generate encrypted code to display to user
-    user_auth = current_user.authentication_options.find_by_credential_type(AuthenticationOption::GOOGLE)
-    @secret_code = Encryption.encrypt_string_utf8(
-      Time.now.strftime('%Y%m%dT%H%M%S%z') + user_auth['authentication_id'] + user_auth['credential_type']
-    )
+    if current_user
+      # Generate encrypted code to display to user
+      user_auth = current_user.authentication_options.find_by_credential_type(AuthenticationOption::GOOGLE)
+      if user_auth
+        @secret_code = Encryption.encrypt_string_utf8(
+          Time.now.strftime('%Y%m%dT%H%M%S%z') + user_auth['authentication_id'] + user_auth['credential_type']
```

Comment:
i know you just moved this from its previous position, but we should be accessing the properties on `user_auth` through their backing methods instead of through string keys:

```ruby
user_auth.authentication_id
# instead of user_auth['authentication_id']
```